tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/alerts/alerts_container_test.ts

```diff
@@ -0,0 +1,163 @@
+import {TimelineModule} from '../timeline/timeline_module';
+
+/** @typehack */ import * as _typeHackStore from '@ngrx/store';
+
+describe('Alerts Container', () => {
+  let store: MockStore<State>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [DebuggerComponent, DebuggerContainer],
+      imports: [
+        AlertsModule,
+        CommonModule,
+        ExecutionDataModule,
+        InactiveModule,
+        StackTraceModule,
+        TimelineModule,
+      ],
+      providers: [
+        provideMockStore({
+          initialState: createState(createDebuggerState()),
+        }),
+        DebuggerContainer,
+      ],
+    }).compileComponents();
+    store = TestBed.get(Store);
+  });
+
+  it('renders number of alerts and breakdown: no alerts', () => {
+    const fixture = TestBed.createComponent(AlertsContainer);
+    store.setState(
+      createState(
+        createDebuggerState({
+          activeRunId: '__default_debugger_runs__',
+          alerts: createAlertsState(),
+        })
+      )
+    );
+    fixture.detectChanges();
+
+    const numAlertsValueElement = fixture.debugElement.query(
+      By.css('.num-alerts-value')
+    );
+    expect(numAlertsValueElement.nativeElement.innerText).toBe('0');
+    const alertTypeNameElements = fixture.debugElement.queryAll(
+      By.css('.alert-type-name')
+    );
+    expect(alertTypeNameElements.length).toBe(0);
+  });
+
+  it('renders number of alerts and breakdown: one alert type', () => {
+    const fixture = TestBed.createComponent(AlertsContainer);
+    store.setState(
+      createState(
+        createDebuggerState({
+          activeRunId: '__default_debugger_runs__',
+          alerts: createAlertsState({
+            numAlerts: 10,
+            alertsBreakdown: {
+              InfNanAlert: 10,
+            },
+          }),
+        })
+      )
+    );
+    fixture.detectChanges();
+
+    const numAlertsValueElement = fixture.debugElement.query(
+      By.css('.num-alerts-value')
+    );
+    expect(numAlertsValueElement.nativeElement.innerText).toBe('10');
+    const alertTypeNameElements = fixture.debugElement.queryAll(
+      By.css('.alert-type-name')
+    );
+    const alertTypeCountElements = fixture.debugElement.queryAll(
+      By.css('.alert-type-count')
+    );
+    expect(alertTypeNameElements.length).toBe(1);
+    expect(alertTypeNameElements[0].nativeElement.innerText).toBe('NaN/∞');
+    expect(alertTypeCountElements.length).toBe(1);
+    expect(alertTypeCountElements[0].nativeElement.innerText).toBe('∞: 10');
+  });
+
+  it('renders number of alerts and breakdown: three alert types', () => {
+    const fixture = TestBed.createComponent(AlertsContainer);
+    store.setState(
+      createState(
+        createDebuggerState({
+          activeRunId: '__default_debugger_runs__',
+          alerts: createAlertsState({
+            numAlerts: 11,
+            alertsBreakdown: {
+              FunctionRecompilesAlert: 5,
```

Comment:
No action required: :(. Because the typing is not sufficient, it did not catch these. `FunctionRecompilesAlert` and `TensorShapeAlert` are not defined in `AlertType` yet there were no type violations.

Reply:
Yeah. I see your point. The difficulty here is that the keys of an object must be a string or number. I added a TODO item for myself to explore tighter typing.